File: src/main/java/onl/tesseract/entity/player/rank/PlayerRank.java
package onl.tesseract.entity.player.rank;

import lombok.Getter;
import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextColor;
import onl.tesseract.tesseractlib.entity.Title;

@Getter
public enum PlayerRank implements Rank{
    APPRENTI("apprenti",Title.APPRENTI, NamedTextColor.GREEN,RankPlot.APPRENTI_PLOT),
    CONCEPTEUR("concepteur",Title.CONCEPTEUR, NamedTextColor.LIGHT_PURPLE,RankPlot.CONCEPTEUR_PLOT),
    CREATEUR("createur",Title.CREATEUR, NamedTextColor.DARK_PURPLE,RankPlot.CREATEUR_PLOT),
    INGENIEUR("ingenieur",Title.INGENIEUR, NamedTextColor.DARK_BLUE,RankPlot.INGENIEUR_PLOT),
    BATISSEUR("batisseur",Title.BATISSEUR, NamedTextColor.BLUE,RankPlot.BATISSEUR_PLOT),
    VIRTUOSE("virtuose",Title.VIRTUOSE,NamedTextColor.AQUA,RankPlot.VIRTUOSEUR_PLOT)

    ;
    private final String permGroup;
    private final Title title;
    private final TextColor color;
    private final RankPlot rankPlot;

    PlayerRank(String permGroup, Title title,TextColor color, RankPlot rankPlot)
    {
        this.permGroup = permGroup;
        this.title = title;
        this.color = color;
        this.rankPlot = rankPlot;
    }

}

File: src/main/java/onl/tesseract/rank/entity/PlayerRank.kt
package onl.tesseract.rank.entity

import lombok.Getter
import net.kyori.adventure.text.format.NamedTextColor
import net.kyori.adventure.text.format.TextColor
import onl.tesseract.core.title.Title
import onl.tesseract.core.title.TitleService
import onl.tesseract.lib.menu.CustomHeadItemBuilder
import onl.tesseract.lib.service.ServiceContainer
import onl.tesseract.util.HeadConstante

private val serv = ServiceContainer[TitleService::class.java]
@Getter
enum class PlayerRank(
    override val permGroup: String,
    override val title: Title,
    override val color: TextColor,
    val headBuilder: CustomHeadItemBuilder,
    val rankPlot: RankPlot,
    val hoursRequired: Long,
) : Rank{
    APPRENTI(
        "default",
        serv.getById("APPRENTI"),
        NamedTextColor.GREEN,
        HeadConstante.teteApprenti,
        RankPlot.APPRENTI_PLOT,
        0),
    CONCEPTEUR(
        "concepteur",
        serv.getById("CONCEPTEUR"),
        NamedTextColor.LIGHT_PURPLE,
        HeadConstante.teteConcepteur,
        RankPlot.CONCEPTEUR_PLOT,
        4),
    CREATEUR(
        "createur",
        serv.getById("CREATEUR"),
        NamedTextColor.DARK_PURPLE,
        HeadConstante.teteCreateur,
        RankPlot.CREATEUR_PLOT,
        16),
    INGENIEUR(
        "ingenieur",
        serv.getById("INGENIEUR"),
        NamedTextColor.DARK_BLUE,
        HeadConstante.teteIngenieur,
        RankPlot.INGENIEUR_PLOT,
        48),
    BATISSEUR(
        "batisseur",
        serv.getById("BATISSEUR"),
        NamedTextColor.BLUE,
        HeadConstante.teteBatisseur,
        RankPlot.BATISSEUR_PLOT,
        168),
    VIRTUOSE(
        "virtuose",
        serv.getById("VIRTUOSE"),
        NamedTextColor.AQUA,
        HeadConstante.teteVirtuose,
        RankPlot.VIRTUOSEUR_PLOT,
        Long.MAX_VALUE);

}